### DISCUSSION TITLE
Planned Feature: OpenAI Whisper API

### DISCUSSION THREAD

**Discussion by ThioJoe** [ThioJoe/Auto-Synced-Translated-Dubs#54] [Ideas]
Creating this thread for any discussion about OpenAI's new Whisper API, because I plan to integrate it at some point: https://techcrunch.com/2023/03/01/openai-debuts-whisper-api-for-text-to-speech-transcription-and-translation/?tpcc=tcplustwitter

In my opinion, Whisper's transcription is second to none at the moment. Until now the Whisper model could only be run locally, but now it can be run through OpenAI's API for an extremely low cost of $0.006 per minute.

So this could be a step towards a 100% fully automatic transcription and translation process.

**Comment by Pythone** [comment]
This could also be a step towards an almost entirely local version, which is what I wanted.

**Reply by ThioJoe** [reply]
Actually the API I was referring to here is the online API of Whisper. You can already run Whisper locally - which is actually what I do myself, just manually. 

In the mean time, I've uploaded the script I made for transcribing a video with whisper locally so you can try it yourself: https://gist.github.com/ThioJoe/e84c4e649857bf0a290b49a2caa670f7

It only transcribes, it doesn't produce timings or anything right now.

**Reply by Pythone** [reply]
> Actually the API I was referring to here is the online API of Whisper. You can already run Whisper locally - which is actually what I do myself, just manually.
> 
> In the mean time, I've uploaded the script I made for transcribing a video with whisper locally so you can try it yourself: https://gist.github.com/ThioJoe/e84c4e649857bf0a290b49a2caa670f7
> 
> It only transcribes, it doesn't produce timings or anything right now.

thanks

**Reply by Kurdakov** [reply]
btw https://github.com/facebookresearch/fairseq/tree/main/examples/mms - good speech engine from facebook
https://ai.facebook.com/blog/multilingual-model-speech-recognition/
